### PR TITLE
Reformatting drivers folder

### PR DESCRIPTION
**Pull request by DerekTang04**
## Description

*What was completed, changed, or updated?* 

The folder structure for drivers was changed. Folder names were also changed to snake case. In doing so, CMake files (build and test) also needed to be changed to reflect the new driver file paths. Furthermore, a TM-specific circular buffer was created to have extra functionality with MavLink messages. The TM-specific circular buffer inherits from the original circular buffer. Finally, some edits were made to the dma_uart_device.(cpp/hpp) and drivers_config.(cpp/hpp) for compatibility with TM.

<br/>

*Why was this done (if applicable)?*

1) Drivers folder structure and case update at the direction of Derek Tang
2) Actual code changes (non Cmake) done to allow driver compatibility with TM
3) Cmake changes required as driver folder structure was changed.

<br/>

## Testing

*What **manual** tests were used to validate the code?*

Firmware and Testing files were successfully compiled

<br/>

*What **unit** tests were used to validate the code?*

N/A

<br/>

## Documentation

*Milestone number and name:*

*Link to Asana task:*

*Link to Confluence documentation:*

<br/>

## Reminders

- [ ] Add reviewers to the PR

- [ ] Mention the PR in the appropriate discord channel

### CODE REVIEW

**Review by StanleyTang17 — COMMENTED** [review]
UART DMA device changes LGTM. The rest is nitpicking.

File: Tools/Firmware/CMakeLists.txt [hunk 1]
```diff
@@ -24,16 +24,27 @@ include_directories(
...
+    ${ROOT_DIR}/Drivers/common/circular_buffer/inc
+    ${ROOT_DIR}/Drivers/common/drivers_config/inc
+    ${ROOT_DIR}/Drivers/common/dma_uart_device/inc
+    ${ROOT_DIR}/Drivers/iwdg_driver/inc
+    ${ROOT_DIR}/Drivers/motor_channel/inc
+    ${ROOT_DIR}/Drivers/rc_receiver/inc
+    ${ROOT_DIR}/Drivers/sensor_fusion/inc
+    ${ROOT_DIR}/Drivers/sensor_fusion/ccontrol/inc
+
+    # BEFORE FILE RESTURCTURING
```
[the remark this answers is not in the document]
Comment (Yaremadzulynsky) [reply Tools/Firmware/CMakeLists.txt:37]: Right. Thx

File: Drivers/common/dma_uart_device/src/dma_uart_device.cpp [hunk 2]
```diff
@@ -1,9 +1,11 @@
-#include "uart_device.hpp"
 #include "main.h"
+#include "dma_uart_device.hpp"
 
-UARTDevice::UARTDevice(UART_HandleTypeDef* huart)
-    :
-    uart_handle_(huart), circular_buf_(buf_, BUFFER_SIZE) {}
+UARTDevice::UARTDevice(UART_HandleTypeDef* huart, CircularBuffer *buf_) : uart_handle_(huart) {
+    //GOTTA FIND A WAY TO ENSURE THAT THE BUFFER SIZE IS THE SAME AS THE BUFFER SIZE IN THE CIRCULAR BUFFER
```
[the remark this answers is not in the document]
Comment (Yaremadzulynsky) [reply Drivers/common/dma_uart_device/src/dma_uart_device.cpp:5]: Thx. I was wondering if they needed to be.

File: TelemetryManager/Inc/CircularBuffer.hpp
[the remark this answers is not in the document]
Comment (Yaremadzulynsky) [reply TelemetryManager/Inc/CircularBuffer.hpp:1]: Completely agree. Would be much more logical.

**Review by StanleyTang17 — APPROVED** [review]
LGTM, assuming it can compile.

**Review by Yaremadzulynsky — APPROVED** [review]

**Pull request merged**